build_docs.py

```diff
@@ -108,7 +108,7 @@ def requirements(self):
         if version_to_tuple(self.sphinx_version) < (4, 5):
             # see https://github.com/python/cpython/issues/91294
             reqs += ["jinja2<3.1"]
-        if version_to_tuple(self.sphinx_version) <= (3, 2, 1):
+        if version_to_tuple(self.sphinx_version) <= (3, 4, 3):
```

Comment:
```suggestion
        if version_to_tuple(self.sphinx_version) <= (3, 5, 4):
```

(nit) Please refer to https://github.com/python/cpython/blob/3.10/Doc/requirements.txt#L7-L9. Sphinx 3.5.4 [was first version with upper bound](https://github.com/python/cpython/issues/89781#issuecomment-1093935244) for docutils dependency.

Reply:
(nit) I made a mistake in the suggestion, the comparator should be `<`, not `<=`.